### PR TITLE
[core] snapshot create idle timeout

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### Purpose

When a source remains idle for the specified timeout duration in the context of watermarking. It triggers the advancement of snapshots, thereby promoting tag creation.

### Tests

TagAutoCreationTest#testWatermarkIdleTimeoutForceCreatingSnapshot

### CODE REVIEW

**Comment by MonsterChenzhuo** [comment]
@yuzelin PTAL,thanks.

**Comment by MonsterChenzhuo** [comment]
@JingsongLi PTAL,thanks.

File: paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java [hunk 1]
```diff
@@ -96,13 +102,29 @@ private boolean isAutoTag(String tag) {
...
+            return checkIdleTimeout();
+        }
         return timeExtractor instanceof ProcessTimeExtractor
                 && (nextTag == null
                         || isAfterOrEqual(
                                 LocalDateTime.now().minus(delay),
                                 periodHandler.nextTagTime(nextTag)));
     }
 
+    private boolean checkIdleTimeout() {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java:115]: inline this method into `forceCreatingSnapshot`, it is hard to understand from this method name `checkIdleTimeout `

File: paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java [hunk 2]
```diff
@@ -96,13 +102,29 @@ private boolean isAutoTag(String tag) {
     }
 
     public boolean forceCreatingSnapshot() {
+        if (timeExtractor instanceof WatermarkExtractor && idlenessTimeout != null) {
+            return checkIdleTimeout();
+        }
         return timeExtractor instanceof ProcessTimeExtractor
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java:108]: if (timeExtractor instanceof WatermarkExtractor) {
   ....
}
else if (timeExtractor instanceof ProcessTimeExtractor) {
  ....
}

**Comment by MonsterChenzhuo** [comment]
@JingsongLi  PTAL,thanks.

File: paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java (hunk 1 above)
Comment (MonsterChenzhuo) [reply paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java]: done.

File: paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java (hunk 2 above)
Comment (MonsterChenzhuo) [reply paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java]: done.

**Comment by MonsterChenzhuo** [comment]
@JingsongLi  PTAL,thanks.

**Pull request merged**